### PR TITLE
chore(RHTAP-2076): get variables using "build_deploy.sh"

### PR DESCRIPTION
**Pull request by ShakedAviv50313**
chore(RHTAP-2076): get variables using "build_deploy.sh"

### CODE REVIEW

File: tasks/get-variables/get-variables.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: get-variables
```
Comment (ernesgonzalez33) [inline tasks/get-variables/get-variables.yaml:5]: @ShakedAviv50313 please change the name to `parse-build-deploy-script` as mentioned in the jira: https://issues.redhat.com/browse/RHTAP-2076

**Review by ernesgonzalez33 — CHANGES_REQUESTED** [review]

**Comment by ernesgonzalez33** [comment]
@ShakedAviv50313 can you check why the coverage test is failing?

File: tasks/parse-build-deploy-script/parse-build-deploy-script.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  params:
+    - default: "source"
+      description: directory in the `output` workspace to clone the repo into.
+      name: subdirectory
+      type: string
+    - name: path-context
+      default: tests/insights-dashboard-master
+      description: path context directory inside the repo
+      type: string
+  results:
```
Comment (ernesgonzalez33) [inline tasks/parse-build-deploy-script/parse-build-deploy-script.yaml:20]: Add the node-build-version variable as well

File: tasks/parse-build-deploy-script/parse-build-deploy-script.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      description: variable for the component's name
+    - name: image
+      type: string
+      description: variable for the component's image
+  steps:
+    - name: parse-build-deploy-script
+      image: registry.access.redhat.com/ubi9/ubi-minimal:9.3-1361.1699548032
+      script: |
+        #!/usr/bin/env sh
+        set -x
```
Comment (ernesgonzalez33) [inline tasks/parse-build-deploy-script/parse-build-deploy-script.yaml:32]: Change it to `set -ex` so the script exits if there is an error

**Review by ernesgonzalez33 — CHANGES_REQUESTED** [review]

File: tasks/parse-build-deploy-script/parse-build-deploy-script.yaml (hunk 2 above)
Comment (ShakedAviv50313) [reply tasks/parse-build-deploy-script/parse-build-deploy-script.yaml:20]: is it a param or should it be retrieved from a script? (it's not in build_deploy.sh)

File: tasks/parse-build-deploy-script/parse-build-deploy-script.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      description: path context directory inside the repo
+      type: string
+  results:
+    - name: component
+      type: string
+      description: variable for the component's name
+    - name: image
+      type: string
+      description: variable for the component's image
+    - name: node_build_version
```
Comment (ernesgonzalez33) [inline tasks/parse-build-deploy-script/parse-build-deploy-script.yaml:27]: For Tekton, it is better to write it this way: `node-build-version`

File: tasks/parse-build-deploy-script/parse-build-deploy-script.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      type: string
+      description: variable for node's version
+    - name: quay_expire_time
```
Comment (ernesgonzalez33) [inline tasks/parse-build-deploy-script/parse-build-deploy-script.yaml:30]: Here the same

**Review by ernesgonzalez33 — CHANGES_REQUESTED** [review]

**Review by ernesgonzalez33 — APPROVED** [review]

**Pull request merged**